.pipelines/e2e.yml

```diff
@@ -21,36 +23,46 @@ jobs:
   - template: ./templates/template-az-cli-login.yml
     parameters:
       azureDevOpsJSONSPN: $(aro-v4-e2e-devops-spn)
-  - script: |
-      az account set -s $AZURE_SUBSCRIPTION_ID
-      export SECRET_SA_ACCOUNT_NAME=$(SECRET_SA_ACCOUNT_NAME)
-      make secrets
-      . secrets/env
-      echo "##vso[task.setvariable variable=RP_MODE]$RP_MODE"
-    displayName: 🔑 Downloading certificates and secrets from storage account
-    name: setEnv
+  - template: ./templates/template-certificates-and-secrets.yml
   - template: ./templates/template-push-images-to-acr.yml
     parameters:
       rpImageACR: $(RP_IMAGE_ACR)
       buildCommand: publish-image-aro
   - script: |
-      set -e
       set -o pipefail
 
-      . secrets/env
-
       az account set -s $AZURE_SUBSCRIPTION_ID
+      export SECRET_SA_ACCOUNT_NAME=aroe2esecrets
+      make secrets
+      . secrets/env
 
       set -x
       . ./hack/e2e/run-rp-and-e2e.sh
-      trap 'set +e; kill_rp; clean_e2e_db' EXIT
+      trap 'set +e; delete_e2e_cluster; kill_rp; clean_e2e_db' EXIT
 
       deploy_e2e_db
-
       run_rp
       validate_rp_running
       register_sub
 
       export CI=true
-      make test-e2e
-  - template: ./templates/template-az-cli-logout.yml
+
+      if make test-e2e;
```

Comment:
If I'm reading this correctly we're always outputting some sort of must-gather tar, even if it's a 0 byte file. Would it be possible to put the must-gather collection in its own step in the pipeline that only runs if the e2e task fails?

https://docs.microsoft.com/en-us/azure/devops/pipelines/process/conditions?view=azure-devops&tabs=yaml

Reply:
You got it, I was able to get around this by using a conditional in the step. This keeps us from having to write a dummy file in cases where e2e succeeds.

I've also played around with having e2e + must-gather creation + must-gather appending in different stages entirely, but it was not fun. Agent VMs are not the same between stages, and e2e resources and environment variables get cleaned up between stages. It makes the most sense to have it all in a single job/stage where environment is preserved throughout.

Reply:
Yep, ADO uses separate instances of whatever for different stages but keeps it for steps in the same stage. It's a tad annoying when you're trying to do some things in parallel.

Reply:
@dem4gus sounds doable, can you take a look at my recent changes and let me know your thoughts?

Reply:
Looks good.